### DISCUSSION TITLE
To get meeting links in flutter mobile app

### DISCUSSION THREAD

**Discussion by shahhardik04** [code4recovery/12-step-meeting-list#1108] [General]
Hello team,

I want to implement this plugin into flutter mobile app, so some of my questions are :-
1) Is it possible to use the wordpress plugin into flutter mobile app.
2) If yes, then how can i fetch the list of meeting links (step by step configure)

**Comment by joshreisner** [comment]
The plugin itself runs on PHP so it needs a server for that, but we have a way to embed just the front-end in a webview using TSML UI:

https://tsml-ui.code4recovery.org/
https://github.com/code4recovery/tsml-ui

You can add some HTML to a screen on your app and customize it to point to a JSON feed or Google Sheet and you should have a working app

@tiptop96 do you have any pointers? Thanks

**Reply by tiptop96** [reply]
Hey!

Unfortunately I haven't done anything on the app side of things, maybe @enderv can give some pointers?

Or did you mean pointers beside the app? If I misunderstood please tell me, I am home with a fever so brain is at like 5% capacity. 😅

**Reply by joshreisner** [reply]
oops im sorry @tiptop96 my mistake! thanks for replying so quickly, feel better!

**Reply by enderv** [reply]
What Josh said about webview :) For our react native app we just doing a webview to the meetings list page (for a demo we are pointed at a wordpress site but final plan is to use a static html file hosted in azure storage) . This allows us to keep the app simple but we can also do updates to the meeting list html/js without needing to update the app itself.

**Comment by kiyote33** [comment]
Hi @shahhardik04, our 12 Step Meeting List WordPress plugin only works natively with WordPress websites. However, we do have a javascript/react product we call [TSML-UI](https://github.com/code4recovery/tsml-ui). That provides a front-end interface. You would need to use/create a JSON feed to populate the meeting list based on the [Meeting Guide Spec](https://github.com/code4recovery/spec).
HTML code similar to the following would then connect the front-end to the JSON.
https://github.com/code4recovery/12-step-meeting-list/discussions/1058#discussioncomment-4792326
**```
<div
  id="tsml-ui"
  data-src="https://**your-domain-name.org**/wp-admin/admin-ajax.php?action=meetings"
  data-mapbox="**pk.eyJ1IjoiZGF2ZWc4MSIsImEiOiJja3IwcnlhZGoxdjlxMzFxaHNydDBza3ByIn0.t-gsVDq-_V5cl_0rW13y6Q**"
  data-timezone="**America/yourTimeZone**"
></div>
```**
<script src="https://tsml-ui.code4recovery.org/app.js" async></script>

Note the following about the above HTML code:

1. data-src is the URL that points to your JSON feed. The one you see there is for the JSON feed produced by a website using our WordPress plugin.  You could drop the json in the root of your site and register in the HTML code with something like  data-src="https://**your-domain-name.org**/example.json".
2. The free Mapbox key should be your own key which you can obtain from https://mapbox.com.
3. Your timezone needs to reflect a major timezone city which relates to your location.
4. The script src is the critical part that points to our TSML-UI code.

I see Josh has already responded here but I've been a little more verbose, so I'll add this on to what he said. Hope it helps...
